c7n/tags.py

```diff
@@ -1020,9 +1020,9 @@ def validate(self):
         return self
 
     def process(self, resources):
-        related_resources = dict(
-            zip(jmespath.search('[].%s' % self.data['key'], resources), resources))
-        related_ids = set(related_resources)
+        related_resources = list(
+            zip(jmespath.search('[].[%s || `false`]|[]' % self.data['key'], resources), resources))
+        related_ids = set([r[0] for r in related_resources])
```

Comment:
needs handling on the not found case (original did as well) .. ie here `related_ids.remove(False)`